llvm/lib/Target/AMDGPU/SIRegisterInfo.td

```diff
@@ -910,22 +916,31 @@ multiclass SRegClass<int numRegs,
       let isAllocatable = 0;
       let BaseClassOrder = !mul(numRegs, 32);
     }
+
+    if hasNull then {
+      def SReg_ # suffix :
+        SIRegisterClass<"AMDGPU", regTypes, 32,
+                      !dag(add, [!cast<RegisterClass>("SReg_" # suffix # "_XNULL"), !cast<Register>("SGPR_NULL" # suffix)], ["", ""])> {
```

Comment:
What is the `["", ""]` for here?

Reply:
The whole thing is to use the !dag operator to create a dag. The 2nd and 3rd arguments are lists with the same length. The 2nd list is a list of values while the 3rd the corresponding names. It seems that the names are not important. That's why they are blank. But in the latest commit, I replaced the two blank names with "RegClass" and "NullReg" respectively.

Using 128b as an example, this creates the dag (add SReg_128:"RegClass", SReg_128_XNULL:"NullReg").

Reply:
OK but I don't see why you need to use `!dag`. You can write it literally as `(add !cast<RegisterClass>("SReg_" # suffix # "_XNULL"), !cast<Register>("SGPR_NULL" # suffix))`.